rest-api/dao/participant_counts_over_time_service.py

```diff
@@ -176,70 +192,163 @@ def get_enrollment_status_sql(self, filters_sql_ps, filters_sql_p):
     # Add macros for GREATEST and LEAST, as they don't work in SQLite
     # Example: master/rest-api/dao/database_utils.py#L50
 
-    sql = """
-      SELECT
-         SUM(registered_cnt * (cnt_day <= calendar.day)) registered_participants,
-         SUM(member_cnt * (cnt_day <= calendar.day)) member_participants,
-         SUM(full_cnt * (cnt_day <= calendar.day)) full_participants,
-         calendar.day
-       FROM
-       (SELECT c2.day cnt_day,
-               registered.cnt registered_cnt,
-               member.cnt member_cnt,
-               full.cnt full_cnt
-            FROM calendar c2
-            LEFT OUTER JOIN
-            (SELECT COUNT(*) cnt,
-                (CASE WHEN (enrollment_status = 1 OR enrollment_status IS NULL) THEN
-                  DATE(p.sign_up_time)
-                  ELSE NULL END) day
+    sql_by_sample_status_time = """
```

Comment:
Have you run this SQL against sandbox or prod to see how well it performs at scale?

Reply:
Yes, it usually return in 1 or 2 secs. And I run it in prod with a time range of 8 month of all data, it will return in secs.